### ISSUE TITLE
[IMPROVEMENT] Implement a full regression suite for ROBOT e2e test cases like the integration test

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#9679]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

Implement

- [x] a full regression suite for ROBOT e2e test cases like the integration test and we can include this suit in the daily runs.
- [x] separate the suites for negative testing and regression testing
- [x] regularly run the full negative testing suite once every two weeks, aligned with sprint image releases.

**Comment by derekbit** [comment]
cc @yangchiu @c3y1huang @khushboo-rancher @innobead

**Comment by c3y1huang** [comment]
Since the @longhorn/qa are migrating the current regression to Robot cases, I suggest we should also start having the suit in the daily runs, like the current regression.

**Comment by innobead** [comment]
For negative testing, it might be hard to have a full regression due to the long execution time? Or we should have a suite for mandatory tests?

**Comment by c3y1huang** [comment]
> or negative testing, it might be hard to have a full regression due to the long execution time? Or we should have a suite for mandatory tests?

Could we separate the suites for negative testing and regression testing?

**Comment by derekbit** [comment]
> > For negative testing, it might be hard to have a full regression due to the long execution time? Or we should have a suite for mandatory tests?
> 
> Could we separate the suites for negative testing and regression testing?

Agree. Currently, we are mixing them together.

**Comment by derekbit** [comment]
In addition, can we also regularly run the full negative testing suite at least once every week or every two weeks? Only running it during the RC stage and attempting to fix issues in a limited time frame is very risky.

**Comment by c3y1huang** [comment]
> In addition, can we also regularly run the **full negative testing suite** at least once every week or every two weeks? Only running it during the RC stage and attempting to fix issues in a limited time frame is very risky.

I agree. I feel negative test failures normally highlight issues that require more time for investigation. Periodical runs help with earlier detection, which buys more time for the developers.

I vote for every two weeks based on the sprint images release.

**Comment by derekbit** [comment]
All tasks are summarized in https://github.com/longhorn/longhorn/issues/9679#issue-2596199827. Thanks.

**Comment by innobead** [comment]
> I suggest running for every two weeks, aligned with sprint image releases.

Sounds good

**Comment by khushboo-rancher** [comment]
>separate the suites for negative testing and regression testing

I marked the task as completed because we follow adding the negative tests and regression tests separately in robot already. - https://github.com/longhorn/longhorn-tests/tree/master/e2e/tests

**Comment by derekbit** [comment]
Already implemented
